presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java

```diff
@@ -59,9 +60,33 @@ public void testEstimatedSize()
                 estimatedSizeAfterCompact));
     }
 
-    private static PagesIndex newPagesIndex(List<Type> types, int expectedPositions)
+    @Test
+    public void testEagerCompact()
+    {
+        List<Type> types = ImmutableList.of(VARCHAR);
+
+        PagesIndex lazyCompactPagesIndex = newPagesIndex(types, 50, false);
+        PagesIndex eagerCompactPagesIndex = newPagesIndex(types, 50, true);
+
+        for (int i = 0; i < 5; i++) {
+            lazyCompactPagesIndex.addPage(somePage(types));
+            eagerCompactPagesIndex.addPage(somePage(types));
+        }
+
+        for (int i = 0; i < 5; i++) {
+            Block lazyCompactBlock = lazyCompactPagesIndex.getChannel(0).get(i);
+            Block eagerCompactBlock = eagerCompactPagesIndex.getChannel(0).get(i);
+            // VARCHAR sequence pages can be compacted
+            assertTrue(eagerCompactBlock.getRetainedSizeInBytes() < lazyCompactBlock.getRetainedSizeInBytes(), "Expect retained size of block in eager compact PagesIndex smaller than lazy compact PagesIndex");
```

Comment:
Why are you operating on `Blocks` rather than comparing `lazyCompactPagesIndex.getEstimatedSize()` with `eagerCompactPagesIndex.getEstimatedSize()` ?

Reply:
By doing this we don't depend on the implementation details of `PagesIndex.getEstimatedSize()`. For example it could be an estimated size in the future.

Reply:
I don't think it can. HBO depends on this to report memory usage, so it needs to be just that. It still can be an estimate, but not something systematically off.

I think this would make tests simpler. If you disagree, I won't insist.

Reply:
Since this is to verify that the test is valid, make this more obvious in the failure message, and move it to the `addPage` loop.

For the actual test part, I think suggestion from @findepi makes sense